### ISSUE TITLE
DS oath fields were removed despite being ignored

### ISSUE THREAD

**Issue by sajildeh** [l-with/terraform-provider-ldap#89]
I faced an issue with module 0.7.0, when I imported and ran apply the fields for Multi OTP in DS were removed while it was under ignore (both were under ignore in TF). Can you please assist further?

The field name: 
- oath2faEnabled
- oathDeviceProfiles

Regards, Saji

**Comment by l-with** [comment]
Please explain what you mean by "DS", describe the steps to reproduces and provide the terraform code. 

Have you tried it with the actual version of the provider?

**Comment by sajildeh** [comment]
Hi, 
thanks for the reply. by DS I mean the LDAP (Apache Directory Studio).
The steps to reproduce is create a user in LDAP and add the 2 fields I mentioned, import it to the module (terraform) and apply the changes. You should see the the values are ignored but after the apply they will be deleted.
Note: after adding the values back to LDAP and run the apply again terraform ignores them.

-> Regarding the try with the new version - I didn't

`provider "ldap-l-with" {
  alias         = "ds"
  host          = ***
  port          = ***
  bind_user     = ***
  bind_password =***
  tls           = true
  tls_insecure  = true
}

resource "ldap_entry" "ldap_user" {
  provider = ldap-l-with.ds
  for_each = local.users_file
  dn       = each.value.entryDN[0]
  data_json = jsonencode({
    objectClass      = tolist(each.value.dsObjectClass)
    cn               = each.value.cn
    sn               = each.value.sn
    departmentNumber = tolist(each.value.departmentNumber)
    description      = each.value.dsDescription
    givenName        = each.value.givenName
    gsaRAC           = tolist(each.value.gsaRAC)
    mail             = each.value.mail
  })
  ignore_attributes = [**"oath2faEnabled", "oathDeviceProfiles"**, "sun-fm-sam2-nameid-info", "entryUUID", "sun-fm-saml2-nameid-infokey", "sunAMAuthInvalidAttemptsData", "collectiveAttributeSubentries", "createTimestamp", "creatorsName", "ds-privilege-name", "ds-pwp-password-policy-dn", "ds-rlim-lookthrough-limit", "ds-rlim-size-limit", "ds-sync-hist", "etag", "hasSubordinates", "isMemberOf", "modifersName", "modifyTimestamp", "numSubordinates", "subschemaSubentry", "sun-fm-saml2-nameid-info", "telephoneNumber", "inetUserHttpURL"]
}`

Regards,

**Comment by l-with** [comment]
I cannot reproduce this with version 0.10.0. I added an entry, imported it in terraform, added ignore_attributes for an attribute, applied the configuration and the attribute  is still in the entry on the server and also in the terraform state.

**Comment by sajildeh** [comment]
The entity had these 2 attributes ("oath2faEnabled", "oathDeviceProfiles") and they stayed after the import and apply?

Regards,

**Comment by l-with** [comment]
> The entity had these 2 attributes ("oath2faEnabled", "oathDeviceProfiles") and they stayed after the import and apply?
> 
> Regards,

No, I used other attributes and OpenLDAP in my test setup and the stayed after import and apply.
To move forward
1. try with the newest version of the provider
2. provide code that gives me a chance to simply reproduce the effect

**Comment by sajildeh** [comment]
I'm using version 0.7.0 and planning to go to the latest version soon. I'm using a lot of variables with my module and all of them are not touched after the apply, with that said the test you did is not valid as you need to add these 2 attributes ("oath2faEnabled", "oathDeviceProfiles") in your ldap and do the process of import-apply to see if they stay or go away. I'm sure they get removed in 0.7.0 but not able to test on my production setup at the moment,

Regards,
Saji Jildeh

**Comment by l-with** [comment]
@sajildeh I would like to comment on the phrase "the test you did is not valid": the terraform provider handles all attributes in the same way, the name of the attributes certainly do not have any influence on the processing logic.

**Comment by sajildeh** [comment]
@l-with That's exactly why I said its not valid. It's suppose to handle all the values in the same way but somehow its not treating these 2 values the same as while they are in the ignore list they still got deleted/removed once I did the apply.
The only way to see this is to add the values in LDAP and try to do an import then apply (on version 0.7.0)

Regards,
Saji

**Comment by l-with** [comment]
@sajildeh Is it possible that the attribute names are misspelled in ignore_attributes? Perhaps the names should be “oauth2faEnabled” and “oauthDeviceProfiles”.
